OSDOCS-11906: adds RN for UDN (#81384)
openshift/openshift-docs: release_notes/ocp-4-17-release-notes.adoc

[id="ocp-4-17-networking_{context}"]
=== Networking

> [id="ocp-4-17-user-defined-network"]

Reviewer: I think IDs in assemblies are supposed to have context in them?

==== New UserDefinedNetwork CRD for creating primary and secondary networks (Technology Preview)

> With {product-title} {version}, users can create multiple networks and declare them as primary or secondary through the new `UserDefinedNetwork` (UDN) custom resource definition (CRD). With UDN, Users can secure traffic between namespaces without configuring and managing complex network policies. For more information, see xref:./networking/multiple_networks/will add later when docs are live.

Reviewer: {version} doesn't replace properly. Should be {product-version} - or really I think RNs should be hard-coded because they're never really true in these instances to ever update into something other than 4.17.

Users -> users

I see your TODO, but just a reminder to replace with the proper xref. Also, I don't think we're consistent about it, but I think the for more info line would look best on its own paragraph instead of smushed into this one. But totally up to you!

```suggestion
With {product-title} {product-version}, users can create multiple networks and declare them as primary or secondary through the new `UserDefinedNetwork` (UDN) custom resource definition (CRD). With UDN, users can secure traffic between namespaces without configuring and managing complex network policies. For more information, see xref:./networking/multiple_networks/will add later when docs are live.
```

> With {product-title} {version}, users can create multiple networks and declare them as primary or secondary through the new `UserDefinedNetwork` (UDN) custom resource definition (CRD). With UDN, Users can secure traffic between namespaces without configuring and managing complex network policies. For more information, see xref:./networking/multiple_networks/will add later when docs are live.

Reviewer: I know the heading says it, but maybe would be worth also mentioning in the paragraph that this is a TP feature?

[id="ocp-4-17-nw-sriov-day1-nics_{context}"]
==== NIC partitioning for SR-IOV devices (Generally Available)

With this update, the ability to enable NIC partitioning for Single Root I/O Virtualization (SR-IOV) devices at install time is Generally Available.

For more information, see xref:../installing/installing_bare_metal/preparing-to-install-on-bare-metal.adoc#nw-sriov-dual-nic-con_preparing-to-install-on-bare-metal[NIC partitioning for SR-IOV devices].

[id="ocp-4-17-nw-sriov-vfs-day2_{context}"]
==== Host network settings for SR-IOV VFs (Generally Available)

With this update, the ability to update host network settings for Single Root I/O Virtualization (SR-IOV) network virtual functions in an existing cluster is Generally Available.

For more information, see xref:../networking/k8s_nmstate/k8s-nmstate-updating-node-network-config.adoc#virt-example-vf-host-services_k8s_nmstate-updating-node-network-config[Node network configuration policy for virtual functions].

[id="ocp-4-17-user-defined-network_{context}"]

> ==== New user defined network segmentation (Technology Preview)

Reviewer 2: ```suggestion
==== User defined network segmentation (Technology Preview)
```
curious on the "new" ?

> ==== New UserDefinedNetwork CRD for creating primary and secondary networks (Technology Preview)

Reviewer 2: ```suggestion
==== User Defined Network Segmentation (Technology Preview)
```

> With {product-title} 4.17, users can create multiple networks and declare them as primary or secondary networks for their workloads through the technology preview of the `UserDefinedNetwork` (UDN) custom resource definition (CRD). With UDN, users can isolate namespaces without configuring and managing complex network policies. For more information, see xref:./networking/multiple_networks/will add later when docs are live.

Reviewer 2: ```suggestion
For more information, see xref:./networking/multiple_networks/will add later when docs are live.
```
this looks weird? but guessing its a doc detail

Reviewer 3: I agree with @tssurya unless this points to what ever documentation we are adding for UDN this looks like reminder to writer :)

Reviewer 4: ```
through the technology preview of the `UserDefinedNetwork` (UDN) custom resource definition (CRD) which is preferred or via network attachment definition.
```
@tssurya if we continue supporting NAD way we shall explicitely mention that as well imo. wdyt

> With {product-title} 4.17, users can create multiple networks and declare them as primary or secondary through the technology preview of the `UserDefinedNetwork` (UDN) custom resource definition (CRD). With UDN, users can secure traffic between namespaces without configuring and managing complex network policies. For more information, see xref:./networking/multiple_networks/will add later when docs are live.

Reviewer 2: ```suggestion
With {product-title} 4.17, users can create multiple networks and declare them as primary or secondary networks for their workloads through the technology preview of the `UserDefinedNetwork` (UDN) custom resource definition (CRD). With UDN, users can isolate namespaces and tenants without configuring and managing complex network policies. For more information, see xref:./networking/multiple_networks/will add later when docs are live.
```

| General Availability
| General Availability

> |`UserDefinedNetwork` CRD

Reviewer 2: ```suggestion
|`UserDefinedNetwork` Segmentation
```

Author: I can do User defined network segmentation or `UserDefinedNetwork` CRD?
